src/platform/silabs/EndpointQueueFilter.cpp

```diff
@@ -0,0 +1,108 @@
+#include "EndpointQueueFilter.h"
+#include <string.h>
+#include <support/CodeUtils.h>
+#include <support/logging/CHIPLogging.h>
+#include <string_view>
+
+namespace chip {
+namespace Inet {
+
+using FilterOutcome = EndpointQueueFilter::FilterOutcome;
+
+namespace {
+
+bool IsValidMdnsHostName(uint8_t * hostName)
+{
+    for (size_t i = 0; i < sizeof(hostName); ++i)
+    {
+        char ch_data = hostName[i];
+        if (!((ch_data >= '0' && ch_data <= '9') || (ch_data >= 'A' && ch_data <= 'F')))
+        {
+            return false;
+        }
+    }
+    return true;
+}
+
+bool IsMdnsBroadcastPacket(const void * endpoint, const IPPacketInfo & pktInfo, const System::PacketBufferHandle & pktPayload)
+{
+    // if the packet is not a broadcast packet to mDNS port, drop it.
+    VerifyOrReturnValue(pktInfo.DestPort == 5353, false);
+#if INET_CONFIG_ENABLE_IPV4
+    ip_addr_t mdnsIPv4BroadcastAddr = IPADDR4_INIT_BYTES(224, 0, 0, 251);
+    VerifyOrReturnValue(pktInfo.DestAddress == Inet::IPAddress(mdnsIPv4BroadcastAddr), false);
+#endif
+    ip_addr_t mdnsIPv6BroadcastAddr = IPADDR6_INIT_HOST(0xFF020000, 0, 0, 0xFB);
+    VerifyOrReturnValue(pktInfo.DestAddress == Inet::IPAddress(mdnsIPv6BroadcastAddr), false);
+    return true;
+}
+
+bool PayloadContainsCaseInsensitive(const System::PacketBufferHandle & payload, const ByteSpan & pattern)
+{
+    if (payload->TotalLength() == 0 || pattern.size() == 0)
+    {
+        return false;
+    }
+
+    if (payload->HasChainedBuffer() || payload->TotalLength() < pattern.size())
+    {
+        return false;
+    }
+
+    std::basic_string_view<uint8_t> payloadView(payload->Start(), payload->TotalLength());
+    std::basic_string_view<uint8_t> patternView(pattern.data(), pattern.size());
```

Comment:
Don't use std::bassic_string_view --> we have CharSpan for this and you use anything to make it case insensitve.

Reply:
Modified the case insensitive search